### ISSUE TITLE
Inference using Keras exported is significantly slower

### ISSUE THREAD

**Issue by iumyx2612** [ultralytics/yolov5#5646]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Detection, Export

### Bug

I use yolov5s model on my Laptop, using a GTX 1050
This is the speed of pytorch model
![image](https://user-images.githubusercontent.com/69593462/141734855-2daf51a7-7ac1-4957-a268-442785b038b9.png)

Then I exported the model to Keras format
This is the speed of Keras model
![image](https://user-images.githubusercontent.com/69593462/141735102-af73d47d-b1e3-4105-934a-965fc05c2ca2.png)

### Environment

- YOLOv5  2021-11-15 torch 1.9.0 CUDA:0 (GeForce GTX 1050, 4096MiB)
- OS: Windows10
- Python: 3.8

### Minimal Reproducible Example

Inference using Pytorch model:
`python detect.py --weights=weights/yolov5s.pt --source=0 --nosave --view-img --imgsz=320 --half` 

Export to Keras:
`python export.py --data=data/coco128.yaml --weights=weights/yolov5s.pt --include=saved_model --imgsz=320` 

Inference using Keras
`python detect.py --weights=weights/yolov5s_saved_model --source=0 --nosave --view-img --imgsz=320 --half` 

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@iumyx2612 I've removed the bug label as bugs are errors that appear and you've exported and run inference successfully with no errors.

Speeds and results will obviously vary by many factors including hardware, software, drivers, model types/backends and quantization. Note that the --half flag only applies to pytorch models.

**Comment by iumyx2612** [comment]
> Speeds and results will obviously vary by many factors including hardware, software, drivers, model types/backends and quantization. Note that the --half flag only applies to pytorch models.
I think the should be relatively the same for both model because they both runs on my laptop, should they not?
I tried to do inference again without --half flag and the speed of Keras model still remains slow

**Comment by zldrobit** [comment]
@iumyx2612 Inference speed varies among deep learning frameworks. For example, here is a table of training throughput from [the PyTorch paper](https://arxiv.org/abs/1912.01703):
![image](https://user-images.githubusercontent.com/3896992/145325580-95fd4659-726e-4ac2-a092-5c5dfd56e577.png)

**Comment by iumyx2612** [comment]
> @iumyx2612 Inference speed varies among deep learning frameworks. For example, here is a table of training throughput from [the PyTorch paper](https://arxiv.org/abs/1912.01703): ![image](https://user-images.githubusercontent.com/3896992/145325580-95fd4659-726e-4ac2-a092-5c5dfd56e577.png)

Wow, I thought TF would have the fastest inference speed. Thanks for the info
